near-sdk/src/types/gas.rs

```diff
@@ -20,6 +20,18 @@ use serde::{de, Deserialize, Deserializer, Serialize, Serializer};
 #[repr(transparent)]
 pub struct Gas(pub u64);
 
+impl Gas {
+    pub fn from_tgas(tgas: u64) -> Gas {
+        Gas::ONE_TGAS * tgas
+    }
+
+    pub const ONE_TGAS: Gas = Gas(u64::pow(10, 12));
+
+    pub fn from_tgas_float(tgas: f64) -> Gas {
+        Gas((Gas::ONE_TGAS.0 as f64 * tgas) as u64)
+    }
```

Comment:
related #368, also my intuition is that this opens up to overflows on the higher ranges when not necessary. My personal opinion is that we should just expose the value of one tera gas and just allow users to do any arithmetic they want externally, as this doesn't seem like a frequent pattern that would be used.

Reply:
I hadn't see that issue.  I like @frol suggestion to use `giga(1_100)` So perhaps I'll add that or even `mega` so allow for more fine grained gas definitions.

Reply:
Instead of using floats, I think strings might be better? similar to https://github.com/near/near-sdk-rs/blob/9d99077c6acfde68c06845f2a1eb2b5ed7983401/near-sdk-sim/src/units.rs#L10 ? also `units.rs` seems like a right place for this?

Reply:
Or we should make a rust version of [units-js](https://github.com/near/units-js)